integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala

```diff
@@ -775,6 +777,13 @@ object CarbonDataRDDFactory extends Logging {
           loadStatus = CarbonCommonConstants.STORE_LOADSTATUS_FAILURE
           logInfo("DataLoad failure")
           logger.error(ex)
+          ex match {
+            case sparkException: SparkException =>
+              if (sparkException.getCause.isInstanceOf[DataLoadingException]) {
+                executorMessage = sparkException.getCause.getMessage
+              }
+            case _ =>
```

Comment:
@Zhangshunyu ...for underscore case also get the message and assign it to variable executorMessage because lets say the control comes to case _ then variable executorMessage will have empty value and message will remain as "DataLoad failure: " which will be incorrect